modules/images/fetchpriority/load.php

```diff
@@ -0,0 +1,49 @@
+<?php
+/**
+ * Module Name: Fetchpriority
+ * Description: Adds `fetchpriority=high` parameter to the non lazy loaded images.
+ * Experimental: Yes
+ *
+ * @since n.e.x.t
+ * @package performance-lab
+ */
+
+/**
+ * Filters an image tag in content to add the fetchpriority attribute if it is not lazy-loaded.
+ *
+ * @since n.e.x.t
+ *
+ * @param string $filtered_image The image tag to filter.
+ * @param string $context        The context of the image.
+ * @return string The filtered image tag.
+ */
+function fetchpriority_img_tag_add_attr( $filtered_image, $context ) {
+
+	if ( 'the_content' !== $context && 'the_post_thumbnail' !== $context ) {
+		return $filtered_image;
+	}
+
+	if ( ! empty( $filtered_image ) && strpos( $filtered_image, 'loading="lazy"' ) === false && strpos( $filtered_image, 'fetchpriority=' ) === false ) {
+		$filtered_image = str_replace( '<img ', '<img fetchpriority="high" ', $filtered_image );
+		remove_filter( 'wp_content_img_tag', 'fetchpriority_img_tag_add_attr' );
```

Comment:
I think for safety reasons we need to rather include the other filter removal call here as well. Essentially, whenever we add `fetchpriority="high"` to any image, we need to remove all other filters that could possibly do it, so that it's always just one image.

Right now, the way the code is written it would be possible in principle for 2 images to have `fetchpriority="high"` when a featured image without `loading` attribute follows a content image without `loading` attribute. Unlikely but possible, and we need to be stricter here than with lazy-loading as preferably really only the LCP image should have `fetchpriority="high"`.
```suggestion
		remove_filter( 'wp_content_img_tag', 'fetchpriority_img_tag_add_attr' );
		remove_filter( 'post_thumbnail_html', 'fetchpriority_filter_post_thumbnail_html' );
```